alephmarcreader/tests/test_Marc21Reader.py

```diff
@@ -49,6 +49,28 @@ def test_date(self):
 
         self.assertEqual(date[0], u'1734.03.12')
 
+    def test_get_standardized_date(self):
+        # Test with precise date
+        marc21_rd = AlephMarc21Reader('alephmarcreader/tests/sample_data/Marc21/000055275.marc')
+
+        std_date = marc21_rd.get_standardized_date()
+        self.assertEqual(len(std_date), 1)
+        self.assertEqual(std_date[0].start_span, u'1734-03-12')
+        self.assertEqual(std_date[0].end_span, False)
+        self.assertEqual(std_date[0].get_standardized_date_string_KNORA(), u'1734-03-12')
+        self.assertEqual(std_date[0].get_standardized_date_string_ISO8601(), u'1734-03-12')
+
+        # Test with time span
+        marc21_rd = AlephMarc21Reader('alephmarcreader/tests/sample_data/Marc21/000234529.marc')
+
+        std_date = marc21_rd.get_standardized_date()
+        self.assertEqual(len(std_date), 1)
+        self.assertEqual(std_date[0].start_span, u'1744')
+        self.assertEqual(std_date[0].end_span, u'1782')
+        self.assertEqual(std_date[0].get_standardized_date_string_KNORA(), u'1744:1782')
```

Comment:
A valid Knora date requires the calendar. Would that always be Gregorian?

Reply:
As I state here:  
https://github.com/dhlab-basel/alephmarcreader/blob/e679ac9e370a61f1aa984cc7f28c0d110ae63a5a/alephmarcreader/abstractalephmarcreader.py#L114-L136  
I always imply gregorian date.

But to be honest, in the Knora API, I didn't really find anything on how exactly I should design this string to look, so it best fits Knora. Therefor I just left it as it was.  
If you want anything changed, let me know.

Generally, I use a very simple date format, that is ISO conformal, but allows for a lot less. I did so, because the only dates we have are letter dates, so nothing fancy can happen there...

But this made me think - should we handle the person's life span as a standardized date aswell? or is that overkill? (We're only working with the GND anyway, right?)

Reply:
A valid Knora date has to contain the calendar: 

https://github.com/dhlab-basel/Knora/blob/c967083e52a8519f6810d9910f50f95b2cdf856a/docs/src/api-v2/Basic.ts#L23

So it it's always Gregorian, just prepend the string :-)

There should actually be a HTML doc page of the API formats online, I will check!

> But this made me think - should we handle the person's life span as a standardized date aswell? or is that overkill? (We're only working with the GND anyway, right?)

It depends. If you want to create resources for persons in Knora, it has to meet the Knora format. If you just use the GND to reference them, this is not necessary.

Reply:
> So it it's always Gregorian, just prepend the string :-)

Done.

> It depends. If you want to create resources for persons in Knora, it has to meet the Knora format. If you just use the GND to reference them, this is not necessary.

I don't see why I would want to do that. And if ever I want to do that after all, then that would be something for v. 2.0

Reply:
ok :-)